src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1055,7 +1055,9 @@ where
                             // If the batch_input frontier has advanced past this batch time, then
                             // we can advance the `global_source_upper`, which will later downgrade
                             // `cap_set`, as we are sure we have seen all messages for this
-                            // `source_upper`.
+                            // `source_upper`. Note that downgrading this capability directly
+                            // communicates to the `persist_sink` that a timestamp (for
+                            // an offset) is closed, and should be `compare_and_appended`.
```

Comment:
@aljoscha please review this carefully...if this is wrong then i clearly don't understand timely